sync/syncer.go

```diff
@@ -172,12 +173,16 @@ func (s *Syncer) Run(ctx context.Context) error {
 				"phase", phaseLabel[doFetch],
 				"more", resp.MoreToFollow,
 				"cursor", resp.Cursor,
-				"devices", len(resp.Devices),
+				// "device-events" because there could be multiple events for the same device.
+				"device-events", len(resp.Devices),
 			}
 			if !resp.FetchedUntil.IsZero() {
 				// these just gunk up the logs if they're zero
 				logs = append(logs, "fetched_until", resp.FetchedUntil)
 			}
+			for _, device := range resp.Devices {
+				logger.Debug("returned-device", fmt.Sprintf("%+v", device))
+			}
```

Comment:
IMO this is handy for debugging. E.g. we've seen multiple events with the same serial number (e.g. if you add+remove+add a device from a "MDM server" in between syncer runs)

Reply:
This might be worthwhile for an optionized additional attribute like `Assigner.debug` bool. I'd be open to that. Especially if one isn't using the built-in assigner (and only the syncer by itself).